.github/actions/upstream-test/action.yml

```diff
@@ -40,14 +40,34 @@ runs:
           username: ${{ github.repository_owner }}
           password: ${{ inputs.token }}
 
+      - name: Configure docker image tag
+        shell: bash -l {0}
+        run: |
+          image_base_tag=ghcr.io/${{ github.repository_owner }}/dagmc-ci-ubuntu-${{ inputs.ubuntu_version }}-${{ inputs.compiler}}-hdf5_${{ inputs.hdf5_version}}-moab_${{ inputs.moab_version }}
+          if [ ${{ inputs.geant4_version }} != "OFF" ] ; then
+            image_base_tag=${image_base_tag}-geant4_${{ inputs.geant4_version }}
+            GEANT4=ON
+          else
+            GEANT4=OFF
+          fi
+          if [ ${{ inputs.double_down_version }} != "OFF" ] ; then
+            image_base_tag=${image_base_tag}-double_down_${{ inputs.double_down_version }}
+            DOUBLE_DOWN=ON
+          else
+            DOUBLE_DOWN=OFF
+          fi
+          echo "image_base_tag=${image_base_tag}" >> $GITHUB_ENV
+          echo "GEANT4=${GEANT4}" >> $GITHUB_ENV
+          echo "DOUBLE_DOWN=${DOUBLE_DOWN}" >> $GITHUB_ENV
+
       - name: Installing Dependencies in Docker image
         uses: firehed/multistage-docker-build-action@v1
         with:
-          repository: ghcr.io/${{ github.repository_owner }}/dagmc-ci-ubuntu-${{ inputs.ubuntu_version }}-${{ inputs.compiler}}-geant4_${{ inputs.geant_version }}-hdf5_${{ inputs.hdf5_version }}-moab_${{ inputs.moab_version }}
+          repository: ${{ env.image_base_tag }}
           stages: base, external_deps, hdf5, moab, dagmc
           server-stage: dagmc_test
           quiet: false
           parallel: true
           tag-latest-on-default: false
           dockerfile: CI/Dockerfile
-          build-args: COMPILER=${{ inputs.compiler }}, geant4_version=${{ inputs.geant_version }}, UBUNTU_VERSION=${{ inputs.ubuntu_version }}, HDF5_VERSION=${{ inputs.hdf5_version }}, MOAB_BRANCH=${{ inputs.moab_version }}, double_down=${{ inputs.double_down }}
+          build-args: UBUNTU_VERSION=${{ inputs.ubuntu_version }}, COMPILER=${{ inputs.compiler }}, HDF5_VERSION=${{ inputs.hdf5_version }}, MOAB_VERSION=${{ inputs.moab_version }}, GEANT4=${{ env.GEANT4 }}, GEANT4_VERSION=${{ inputs.geant4_version }}, DOUBLE_DOWN=${{ env.DOUBLE_DOWN }}, DOUBLE_DOWN_VERSION=${{ inputs.double_down_version }}
```

Comment:
If it is allowed (I think it is), maybe it's best to list these arguments one per line so that future changes can be more easily identified

Reply:
I will try